packages/qgrid-core/src/scene/render/render.d.ts

```diff
@@ -5,7 +5,7 @@ import { RenderStrategy } from './render.strategy';
 
 export declare class Renderer {
   defaultStrategy: RenderStrategy;
-  readonly rows: { left: any[]; right: any[]; mid: any[] };
+  readonly rows: Record<string, any>;
```

Comment:
I'd create an interface for `left, right, mid` object and use it here.
```typescript
export interface RowsPosition {
  left: any[];
  mid: any[];
  right: any[];
}
....
readonly rows: RowsPosition;
```
Because `Record<string, ...>` could accept any string as a key;